### PR TITLE
WIP user detail api

### PR DESCRIPTION
**Pull request by jimabramson**
@dan-f @BenjiLee @tlindaliu 

Working on PUT impl/tests now, but ready for feedback on what's here

### CODE REVIEW

File: discussions/api/v1/serializers.py [hunk 1]
```diff
@@ -7,8 +7,33 @@ class Meta:
         model = User.ReadState
 
 
-class UserSerializer(EmbeddedDocumentSerializer):
-    read_states = ReadStatesSerializer(many=True)
+class UserSerializer(DocumentSerializer):
 
     class Meta:
         model = User
+        fields = ('id', 'username', 'external_id', 'default_sort_key')
```
Comment (BenjiLee) [inline discussions/api/v1/serializers.py:16]: Where does `id` come from?

Comment (tlindaliu) [inline discussions/api/v1/serializers.py:16]: did you mean `_id`?

Comment (jimabramson) [inline discussions/api/v1/serializers.py:16]: "id" is what the caller sees (sometimes) but "_id" is what i mean.  will fix

Comment (jimabramson) [inline discussions/api/v1/serializers.py:16]: actually it breaks if i do that.   i believe mongoengine maps "id" to "_id" when needed, or maybe i'm just getting lucky?

**Comment by jimabramson** [comment]
updated with PUT support and tests.  I'll see if I can fix the travis issue real quick.

File: discussions/api/v1/serializers.py (hunk 1 above)
Comment (BenjiLee) [inline discussions/api/v1/serializers.py:16]: It's also defined as a read_only_field if you used _id

**Pull request merged**